src/rebar_git_resource.erl

```diff
@@ -11,6 +11,9 @@
 
 -include("rebar.hrl").
 
+%% Regex used for parsing scp style remote url
+-define(SCP_PATTERN, "\\A(?<username>[^@]+)@(?<host>[^:]+):(?<path>.*)(\\.git)?\\z").
```

Comment:
user@host:patha/pathb and user@host:patha/pathb.git are both valid scp style urls for git. So I've taken the match off .git all together as filename:rootname takes care of making those the same when they're matched against.